src/main/scala/strawman/collection/package.scala

```diff
@@ -84,23 +86,52 @@ import strawman.collection.mutable.{ArrayBuffer, Buildable}
  *         map, flatMap, ++, zip
  */
 package object collection extends LowPriority {
-  import scala.language.implicitConversions
+
   // ------------------ Decorators to add collection ops to existing types -----------------------
 
   /** Decorator to add collection operations to strings. */
   implicit def stringToStringOps(s: String): StringOps = new StringOps(s)
 
-  /** Decorator to add collection operations to arrays. */
+  /** Decorator to add collection operations to arrays.
+   *  Note: should be specialized for efficiency.
+   */
   implicit def arrayToArrayOps[A](as: Array[A]): ArrayOps[A] = new ArrayOps[A](as)
+
+  implicit def immutableArrayToArrayOps[A](as: immutable.Array[A]): immutable.ArrayOps[A] = new immutable.ArrayOps[A](as)
```

Comment:
Why do we need to go through an implicit conversion here rather than making `immutable.Array` extend `immutable.ArrayOps`?